processor/processor.go

```diff
@@ -2426,16 +2435,11 @@ func ConvertToFilteredTransformerResponse(events []transformer.TransformerEventT
 				supportedMessageTypesCache[event.Destination.ID] = supportedTypes
 			}
 			if supportedTypes.ok {
-				messageType, typOk := event.Message["type"].(string)
-				if !typOk {
-					// add to FailedEvents
-					errMessage = "Invalid message type. Type assertion failed"
-					resp = transformer.TransformerResponseT{Output: event.Message, StatusCode: 400, Metadata: event.Metadata, Error: errMessage}
-					failedEvents = append(failedEvents, resp)
-					continue
-				}
-				messageType = strings.TrimSpace(strings.ToLower(messageType))
-				if !slices.Contains(supportedTypes.values, messageType) {
+				allow, failedEvent := eventfilter.AllowEventToDestTransformation(event, supportedTypes.values)
```

Comment:
This is called only if the destination's configuration has a `supportedMessageTypes` field. Otherwise the assumption is that all types are supported and this block is skipped.
Is this condition valid for the message type filtering scenario that is being introduced too?

Reply:
The logic was that way even before I added my changes. I have let it be like that for now

Reply:
I think we were ok with that approach and let it be that way

Reply:
Yes, the previous behavior was the desired one and we need to keep it that way.
The question is whether this new kind of message type filtering should only be applied only to destinations that have `supportedMessageTypes` in their config or to every destination.

Reply:
IMHO, Since the message type filtering deals with the hybrid-mode destination's cloud behaviour(in terms of filtering events), we can keep the previous behaviour as is i.e only those destination(s) which have `supportedMessageTypes` would be eligible for this kind of event filtering logic.

@krishna2020 to confirm on this